### DISCUSSION TITLE
Assertspecial Easing Effects

### DISCUSSION THREAD

**Discussion by shinra358** [ikemen-engine/Ikemen-GO#1935] [Feature Request]
### Is your feature request related to a problem? Please describe.

All AS flag effects happen instantly and theres no way to adjust this

### Describe the solution you'd like

I would like easing effects added to the assertspecial sctrl.

https://learn.microsoft.com/en-us/dotnet/maui/user-interface/animation/easing?view=net-maui-8.0

https://learn.microsoft.com/en-us/dotnet/desktop/wpf/graphics-multimedia/easing-functions?view=netframeworkdesktop-4.8

That way, if something for example has a nofg flag, it will ease invisibility instead of abruptly disappear (slowly fade). And when the effect ends, it will ease back in (slowly fade back in). Would affect every flag that deals with invisibility.

type = assertspecial
trigger1 = 
flag1 = 
ease = (back, bouce, circle, cubic, etc) , (in, out, inout)

### Additional context

time that the flag is activated will automatically be inserted into the equation by the engine so it'll know when to start the shifting of the effect

**Comment by potsmugen** [comment]
All AssertSpecials are booleans (1 or 0) so that probably won't happen to be honest. I think what you're looking for is more like a way to change the stage transparency.

**Comment by shinra358** [comment]
nah, just a way to make it not immediately disappear. like if a person does a super, lots of chars have no fg flags to be able to fully see what design they have for the bg. Having that fg have an easing fade instead of an abrupt one would be a more stylish and visually appealing choice. whether it be stage or character. but stage effects alone, this is already achievable through animations and char/stage interactions. It just isnt achievable through assertspecial. and creators dont have time to code stage interactions with every stage to be able to achieve this without assertspecial because it's kinda impossible because of the vast amounts of created stages.

if one doesnt want to make options for easing, one could choose 1 easing choice and just make it default without any user selectable options and have it replace the abrupt disappearing internally.

bgpalfx is the closest thing that could remedy the situation but theres no way to target a specific element without that. it's all or nothing with bgpalfx
